openassessment/assessment/api/peer.py

```diff
@@ -59,6 +59,33 @@ def submitter_is_finished(submission_uuid, peer_requirements):
         raise PeerAssessmentRequestError(u'Requirements dict must contain "must_grade" key')
 
 
+def _has_enough_assessment(total_assessments, workflow_created_at, peer_requirements):
+    """
+    Check if a workflow has enough number of peer assessments.
+
+    Args:
+        total_assessments (int): Total number of assessment received on a workflow
+        workflow_created_at (Datetime): Workflow creation time
+        peer_requirements (dict): Dictionary with the key "must_grade" indicating
+            the required number of submissions the student must grade.
+
+    Returns:
+        bool
+    """
+
+    flexible_grading_days = peer_requirements.get('flexible_grading_days', 7)
+    flexible_grading_percentage = peer_requirements.get('flexible_grading_graded_by_percentage', 30)
+
+    required_peer_grades = peer_requirements["must_be_graded_by"]
+
+    if peer_requirements.get("enable_flexible_grading"):
+        days_elapsed = (timezone.now().date() - workflow_created_at.date()).days
```

Comment:
I'm not 100% sure that `workflow_created_at.date()).days` is what we're looking for. Is this the date the user submitted the text for peer review?

Reply:
There is a ``submitted_date`` field in [edx/edx-submission](https://github.com/edx/edx-submissions/blob/master/submissions/models.py#L336) repo. Also the workflow gets created upon submission. I think its more correct to use ``submitted_date``, but it requires an additional query.

Reply:
@shimulch No problem with another query, afaik the grade calculation is done when the student or instructor actually access the view. 
I'm not sure there's a cron job anywhere to recalculate grades, but even in this case I don't see a problem in using an extra query.